### DISCUSSION TITLE
Add support for middleware rewrite proxy headers

### DISCUSSION THREAD

**Discussion by johnsardine** [vercel/next.js#33200] [Ideas]
### Describe the feature you'd like to request

I would like to be able to specify headers to be sent by the proxy that is running behind the `NextResponse.rewrite` middleware response.

I understand it's possible to set headers in the `NextResponse.rewrite` response, but these headers are sent to the user visiting the next.js page, what I would like is to set headers that the destination server will receive.

Upon digging into the next.js internals, I understand we'd need to provide such headers in the `Proxy` instance within `proxyRequest` inside `/packages/next/server/base-server.ts`

### Describe the solution you'd like

I would like to be able to, perhaps, provide a config object as second parameter of `NextResponse.rewrite` which could accept something like `forwardHeaders` that accepts a Headers object and this would then be merged with the headers that the proxy may already need to send.

### Describe alternatives you've considered

Currently, my alternative solution is preform this rewrite using nginx and provide the headers there. If these headers can be forwarded within nextjs, I believe this feature will be even more powerful.

**Comment by johnsardine** [comment]
It appears this would fix the problem raised in this discussion https://github.com/vercel/next.js/discussions/31188

**Comment by manoj-mukherjee-maersk** [comment]
any solution? Are we supporting nextjs middleware as http-proxy?

**Comment by icyJoseph** [comment]
Gonna have to vouch for this one as well.

**Comment by JoshBarr** [comment]
I've also run into this limitation. I've raised a PR: https://github.com/vercel/next.js/pull/35093

**Comment by yoniheib** [comment]
Would be great if this was supported.  I'd also like to proxy an authorization header to the destination without exposing it to the client.

**Comment by lockinmarv** [comment]
Any update on this?

**Comment by audriuiv** [comment]
Any updates?
